search.py

```diff
@@ -6,52 +6,82 @@
 # Generic Functions
 
 
-def prepare_tree(xml_file):
-    """create etree and root from the given xml file"""
-    tree = etree.parse(xml_file)
+def prepare_tree(xml_file_path):
+    """create etree and root from the given xml file
+    Arguments: xml_file_path [string]: string of absolute or relative file URL
+    Return: tree [etree]: etree of XML file element objects
+            root [element] root element of tree
+    """
+    tree = etree.parse(xml_file_path)
     root = tree.getroot()
     return tree, root
 
 
-def tree_to_list(root):
+def root_to_list(root):
+    """takes in etree root node, parses it to a depth-first ordered list
+    Arguments: root
+    Returns: List of element objects in depth-first order
+    """
```

Comment:
You need to give the type and a short description of the argument and return values. Keep the heading ("Arguments", "Returns") on separate lines from the description, eg.:

```python

def root_to_list(root):
    """takes in etree root node, parses it to a depth-first ordered list

    Arguments:
        root (FIXME type): FIXME description

    Returns:
        [FIXME type]:  List of element objects in depth-first order
    """

This applies to all other function docstrings you write.

```